### DISCUSSION TITLE
Build arm images using a loop device

### DISCUSSION THREAD

**Discussion by jimmykarily** [kairos-io/kairos#1437] [Announcements]
[The way we build arm images](https://github.com/kairos-io/osbuilder/blob/master/tools-image/build-arm-image.sh) is very different from the way we install from an iso. This leads to bugs like this: https://github.com/kairos-io/kairos/issues/622

We should create the arm images by running the installation like we would run it from an live cd. We can achieve this if we mount the final image on a loop device and perform the installation against that. This will allow us to get rid of that script along with hacks like the fix for the issue above.

**Comment by jimmykarily** [comment]
Move this to a discussion. Let's collect all the thoughts here to at least remember why we didn't go down this road. It seems to be more complicated that we thought. E.g.

- If we chroot to be able to run the installer, we need to do that in an emulator (otherwise some commands won't work). Doing that from withing kuberentes will be tricky (e.g. through the osbuilder controller).

**Comment by mudler** [comment]
random thought: we can have a separate stage "flash" that is meant to be called during flashing with some specific, opinionated setup. For instance, only static copy of files is supported, and to a specific folder (where the rootfs is mounted).

In that way:
- we keep the process arch-agnostic
- we consolidate the business logic in one place only
